.devcontainer/Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM rust:bookworm
```

Comment:
I think we should document that this exists / how to use dev containers in the documentation. Perhaps at 

https://datafusion.apache.org/contributor-guide/getting_started.html#getting-started

We can do it as a follow on PR